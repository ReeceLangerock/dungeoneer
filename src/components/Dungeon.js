import React from "react";
import styled from "styled-components";

// const FooterStyled = styled.div``;

// const I = styled.i``;

export class Dungeon extends React.Component {
  render() {
    return  <div/>;
  }
}
export default Dungeon;
