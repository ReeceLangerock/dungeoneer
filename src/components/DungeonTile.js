import React from "react";
import styled from "styled-components";

// const FooterStyled = styled.div``;

// const I = styled.i``;

export class DungeonTile extends React.Component {
  render() {
    return <div/>;
  }
}
export default DungeonTile;
