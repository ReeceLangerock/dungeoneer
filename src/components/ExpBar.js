import React from "react";
import styled from "styled-components";

const CurrentHealthContainer = styled.div``;

const MaxHealthContainer = styled.div``;

export class ExpBar extends React.Component {
  render() {
    return (
      <div>
        <MaxHealthContainer>
          <CurrentHealthContainer />
        </MaxHealthContainer>
      </div>
    );
  }
}
export default ExpBar;
