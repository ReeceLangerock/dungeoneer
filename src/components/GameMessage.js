import React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import * as actions from './../redux/actions/actions'

const Message = styled.div`
font-family: 'OptimusPrinceps';
color: #7f1517;
background: #191919;
flex-grow: 1;
display: flex;
font-size: 5rem;
justify-content: center;
flex-direction: column;
align-items: center;
z-index:10;
transition: 1s all fade-in-out;
`

const Button = styled.button`
padding: 10px 10px;
border: 2px solid #7f1517;
color: #7f1517;
font-size: 1.5rem;
background: #191919;
cursor: pointer;

:hover {
  background: #7f1517;
  
  color: #191919;
  transition: .5s ease-in-out;
}
`

export class GameMessage extends React.Component {
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
  }
  handleClick () {
    this.props.updateGameStatus('init')
  }
  render () {
    return (
      <Message>
        {this.props.message}
        {this.props.message === 'You Died' && <Button onClick={this.handleClick}>Try Again?</Button>}
        {this.props.message === 'Victory!' && <Button onClick={this.handleClick}>Play Again?</Button>}
      </Message>
    )
  }
}

// const mapStateToProps = state => ({})

export default connect(null, actions)(GameMessage)
