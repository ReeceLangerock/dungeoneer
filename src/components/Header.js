import React from "react";
import styled from "styled-components";

// const FooterStyled = styled.div``;

// const I = styled.i``;

export class Header extends React.Component {
  render() {
    return  <div/>;
  }
}
export default Header;
