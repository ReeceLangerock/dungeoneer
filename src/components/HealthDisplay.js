import React from "react";
import styled from "styled-components";

const CurrentHealthContainer = styled.div`
  position: absolute;
  top: 0px;
  width: ${props => props.healthPercentage}%;
  z-index: -3;
  background-color: green;
  height: 100%;
`;

const MaxHealthContainer = styled.div`
  font-size: 1rem;

  padding: 2px 0px 2px 0px;
  position: relative;
  overflow: hidden;

  width: 90%;
  border: 1px solid;
  border-radius: 10px;
  margin: 0 auto;
`;

export class HealthDisplay extends React.Component {
  render() {
    const healthPercentage = this.props.currentHealth / this.props.maxHealth * 100;
    return (
      <div>
        <MaxHealthContainer>
          {this.props.currentHealth} / {this.props.maxHealth}
          <CurrentHealthContainer healthPercentage={healthPercentage} />
        </MaxHealthContainer>
      </div>
    );
  }
}
export default HealthDisplay;
