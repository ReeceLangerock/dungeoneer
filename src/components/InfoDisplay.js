import React from "react";
import styled from "styled-components";

const TextContainer = styled.div`
  border: 1px solid;
  padding: 3px;
  height: 120px;
`;

const InfoText = styled.h5`
  margin: 5px 0px 5px 0px;
  text-align: left;
  font-weight: normal;
  font-size: 1rem;
`;

export class InfoDisplay extends React.Component {
  render() {
    return (
      <TextContainer>
        <InfoText>weapon: {this.props.weapon}</InfoText>
        <InfoText>attack: {this.props.attack}</InfoText>
        <InfoText>level: {this.props.level}</InfoText>
      </TextContainer>
    );
  }
}
export default InfoDisplay;
