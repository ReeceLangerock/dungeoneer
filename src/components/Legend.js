import React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import * as actions from './../redux/actions/actions'

const Display = styled.span`
position: absolute;
left: 50%;
transform: translate(-50%, -50%);
width: ${window.innerWidth * 0.75 > 300 ? '300px' : '80%'};

top: 40%;
z-index: 8;
background: white;
border: 3px solid black;
cursor: pointer;
color: #7f1517;
padding: 5px;
box-shadow: 0 14px 28px rgba(255,255,255,0.25), 0 10px 10px rgba(255,255,255,0.22);
transition: .75s ease-in-out;
opacity: ${props => (props.info ? 100 : 0)};
cursor: ${props => (props.info ? 'pointer' : 'default')};

h1 {
  font-family: 'OptimusPrinceps';
  font-weight: normal;  
  font-size: 3rem;
  margin: 0px;
}

`
const Icon = styled.i`
color: ${props => props.color};
`
const LegendRow = styled.div`
padding: 1px 5px;
Display: flex;
font-size: 16px;
color: black;

justify-content: space-between;
`
export class Legend extends React.Component {
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
  }
  handleClick () {
    if (this.props.info) {
      this.props.toggleInfo()
    }
  }
  render () {
    return (
      <Display onClick={this.handleClick} info={this.props.info}>
        <h1>How To Play</h1>
        <p>Use arrow keys or WASD to move your player. Defeat the boss lurking in the 4th level of the dungeon to win! </p>
        <hr />
        <h1>Legend</h1>

        <LegendRow>Player: <Icon className='fa fa-square' aria-hidden='true' color='#0000BA' /></LegendRow>
        <LegendRow>Enemy: <Icon className='fa fa-square' aria-hidden='true' color='#7f1517' /></LegendRow>
        <LegendRow>Health: <Icon className='fa fa-square' aria-hidden='true' color='#315C2B' /></LegendRow>
        <LegendRow>Weapon: <Icon className='fa fa-square' aria-hidden='true' color='#B9A938' /></LegendRow>
        <LegendRow>Portal: <Icon className='fa fa-square' aria-hidden='true' color='#701C6F' /></LegendRow>
      </Display>
    )
  }
}
const mapStateToProps = state => ({
  info: state.dungeonReducer.info
})
export default connect(mapStateToProps, actions)(Legend)
