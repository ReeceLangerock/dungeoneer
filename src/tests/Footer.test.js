import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, mount, find, render, simulate } from 'enzyme'
import enzymeSerializer from 'enzyme-to-json/serializer'

import { Footer } from './../components/Footer'

expect.addSnapshotSerializer(enzymeSerializer)
describe('Footer', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Footer />, div)
  })

  let component
  let props

  beforeEach(() => {
    component = mount(<Footer />)
  })

  it('matches snapshot', () => {
    const comp = shallow(<Footer />)
    expect(comp).toMatchSnapshot()
  })
})
