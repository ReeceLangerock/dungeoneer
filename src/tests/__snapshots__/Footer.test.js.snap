// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<styled.div
  className="Footer"
>
  Â© 
  2017
   
  <a
    href="http://reecelangerock.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    Reece Langerock
  </a>
  <styled.i
    aria-hidden="false"
    className="fa fa-github"
    onClick={[Function]}
  />
</styled.div>
`;
